prime-router/metadata/hl7_mapping/OML_O21/base/service-request-order-observation.yml

```diff
@@ -5,6 +5,6 @@ elements:
     schema: ../../common/observation-request
 
   - name: order-observation-date-time-start
-    condition: 'Bundle.entry.resource.ofType(Specimen).collection.collected.exists()'
-    value: [ 'Bundle.entry.resource.ofType(Specimen).collection.collected' ]
+    condition: '%rootResource.entry.resource.ofType(Specimen).collection.collected.exists()'
```

Comment:
TLDR; This is a side effect of no longer passing in `Bundle` as the context.  `%rootResource` is another reserved constant that should be used instead.

[This](https://github.com/hapifhir/org.hl7.fhir.core/blob/64f034ebbff80037d0fa24ef0f96031b02956d0a/org.hl7.fhir.r4/src/main/java/org/hl7/fhir/r4/utils/FHIRPathEngine.java#L803) array passed into the execution now contains the context and not bundle and the downstream implication is that [this](https://github.com/hapifhir/org.hl7.fhir.core/blob/64f034ebbff80037d0fa24ef0f96031b02956d0a/org.hl7.fhir.r4/src/main/java/org/hl7/fhir/r4/utils/FHIRPathEngine.java#L3084) while loop no longer finds `Bundle`

Reply:
A couple comments/questions on this:

1. We should also document `rootResource` as a reserved/special constant.
2. How do these work? I don't see any special handling for them on our end, so I guess they were already doing this in the library? If you could point me in the right direction for documentation or code that explains it, I'd appreciate it.
3. Should we add anything to prevent folks from explicitly assigning values to the reserved constants? Is that already handled by the library as well? Not sure what would take precedence in those cases.

Reply:
- Yep, I'll add another note on it (actually I'm going to break out a new section and enumerate all of them)
- The code in question is [here](https://github.com/hapifhir/org.hl7.fhir.core/blob/64f034ebbff80037d0fa24ef0f96031b02956d0a/org.hl7.fhir.r4/src/main/java/org/hl7/fhir/r4/utils/FHIRPathEngine.java#L1711)
- The way the code is written is that setting `%context` (or any other reserved constant) would fail silently as the the hosts constant resolver is invoked last.  It's a good idea to update our constant adding code to throw an exception if someone tried to do it

Reply:
Sounds good, that third item can be a separate story if it wouldn't make sense to add here, sounds like we're no worse off than we were before. The rest of that makes sense, thanks!

Reply:
It wasn't too much work so I threw it in as part of this work